diff --git a/cmd/admin_auth_oauth.go b/cmd/admin_auth_oauth.go
index c4a8619..c8b7620 100644
--- a/cmd/admin_auth_oauth.go
+++ b/cmd/admin_auth_oauth.go
@@ -34,9 +34,10 @@ func oauthCLIFlags() []cli.Flag {
 			Usage: "Client ID (Key)",
 		},
 		&cli.StringFlag{
-			Name:  "secret",
-			Value: "",
-			Usage: "Client Secret",
+			Name:    "secret",
+			Value:   "",
+			Usage:   "Client Secret",
+			Sources: cli.EnvVars("GITEA_ADMIN_OAUTH2_SECRET"),
 		},
 		&cli.StringFlag{
 			Name:  "auto-discover-url",
